Source/WebCore/PAL/pal/text/TextCodecSingleByte.cpp

```diff
@@ -180,22 +180,21 @@ template<const SingleByteDecodeTable& decodeTable> SingleByteEncodeTable tableFo
 {
     // Allocate this at runtime because building it at compile time would make the binary much larger and this is often not used.
     static constexpr auto size = std::size(decodeTable) - std::count(std::begin(decodeTable), std::end(decodeTable), replacementCharacter);
-    static const SingleByteEncodeTableEntry* entries;
+    static LazyNeverDestroyed<std::array<SingleByteEncodeTableEntry, size>> entries;
     static std::once_flag once;
     std::call_once(once, [&] {
-        auto* mutableEntries = new SingleByteEncodeTableEntry[size];
+        entries.construct();
         size_t j = 0;
-        for (uint8_t i = 0; i < std::size(decodeTable); i++) {
+        for (size_t i = 0; i < std::size(decodeTable); ++i) {
             if (decodeTable[i] != replacementCharacter)
-                mutableEntries[j++] = { decodeTable[i], i + 0x80 };
+                entries.get()[j++] = { decodeTable[i], i + 0x80 };
         }
         ASSERT(j == size);
-        auto collection = WTF::makeIteratorRange(&mutableEntries[0], &mutableEntries[size]);
+        auto collection = WTF::makeIteratorRange(&entries.get()[0], &entries.get()[size]);
```

Comment:
Why do pointers need to be involved? Isn't a `span` a collection we can pass to the functions?

Reply:
`WTF::makeIteratorRange()` seems to take 2 iterators at the moment, no container. But I can fix that.